src/RpcServer/RpcServer.Blockchain.cs

```diff
@@ -280,15 +280,23 @@ protected virtual JToken GetCandidates(JArray _params)
             {
                 script = sb.EmitDynamicCall(NativeContract.NEO.Hash, "getCandidates", null).ToArray();
             }
-            using ApplicationEngine engine = ApplicationEngine.Run(script, snapshot, settings: system.Settings, gas: settings.MaxGasInvoke);
+            StackItem[] resultstack;
+            try
+            {
+                using ApplicationEngine engine = ApplicationEngine.Run(script, snapshot, settings: system.Settings, gas: settings.MaxGasInvoke);
+                resultstack = engine.ResultStack.ToArray();
+            }catch
+            {
+                throw new RpcException(RpcError.InternalServerError.WithData("Can't get candidates."));
+            }
+
             JObject json = new();
             try
             {
-                var resultstack = engine.ResultStack.ToArray();
                 if (resultstack.Length > 0)
                 {
                     JArray jArray = new();
-                    var validators = NativeContract.NEO.GetNextBlockValidators(snapshot, system.Settings.ValidatorsCount);
+                    var validators = NativeContract.NEO.GetNextBlockValidators(snapshot, system.Settings.ValidatorsCount) ?? throw new RpcException(RpcError.InternalServerError.WithData("Can't get next block validators."));
```

Comment:
And also need to do the same with `catch (InvalidOperationException)` below? In order to completely remove the `exception` field from the response.

Reply:
@Jim8y, this one is not addressed yet. I think that the following block of this function isn't relevant anymore:
```
 catch (InvalidOperationException)
            {
                json["exception"] = "Invalid result.";
            }
```

Reply:
Done